nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ListenSMTP.java

```diff
@@ -0,0 +1,385 @@
+            .displayName("Client Auth")
+            .description("The client authentication policy to use for the SSL Context. Only used if an SSL Context Service is provided.")
+            .required(false)
+            .allowableValues(SSLContextService.ClientAuth.NONE.toString(), SSLContextService.ClientAuth.REQUIRED.toString())
+            .build();
+
+    @Override
+    protected Collection<ValidationResult> customValidate(final ValidationContext validationContext) {
+        final List<ValidationResult> results = new ArrayList<>();
+
+        final String clientAuth = validationContext.getProperty(CLIENT_AUTH).getValue();
+        final SSLContextService sslContextService = validationContext.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+
+        if (sslContextService != null && StringUtils.isBlank(clientAuth)) {
+            results.add(new ValidationResult.Builder()
+                    .explanation("Client Auth must be provided when using TLS/SSL")
+                    .valid(false).subject("Client Auth").build());
+        }
+
+        return results;
+
+    }
+
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Extraction was successful")
+            .build();
+
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> propertyDescriptors;
+    private volatile LinkedBlockingQueue<SmtpEvent> messages;
+
+    private volatile SMTPServer server;
+    private AtomicBoolean initialized = new AtomicBoolean(false);
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+        final List<PropertyDescriptor> props = new ArrayList<>();
+        props.add(SMTP_PORT);
+        props.add(SMTP_HOSTNAME);
+        props.add(SMTP_MAXIMUM_CONNECTIONS);
+        props.add(SMTP_TIMEOUT);
+        props.add(SMTP_MAXIMUM_MSG_SIZE);
+        props.add(SSL_CONTEXT_SERVICE);
+        props.add(CLIENT_AUTH);
+        this.propertyDescriptors = Collections.unmodifiableList(props);
+
+    }
+
+    final ComponentLog logger = getLogger();
+
+    // Upon Schedule, reset the initialized state to false
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        initialized.set(false);
+    }
+
+    protected synchronized void initializeSMTPServer(final ProcessContext context) throws Exception {
+        if (initialized.get()) {
+            return;
+        }
+
+        messages = new LinkedBlockingQueue<>(1024);
```

Comment:
I'll make in configurable, however I hope I managed to address the main data loss scenario.